route/v1/stable.route.testsuite.yaml

```diff
@@ -20,3 +20,354 @@ tests:
+      apiVersion: route.openshift.io/v1
+      kind: Route
+      metadata:
+        labels:
+          type: sharded
+        name: hello-openshift-actions
+        namespace: hello-openshift
+      spec:
+        subdomain: hello-openshift
+        tls:
+          termination: edge
+        to:
+          kind: Service
+          name: hello-openshift
+        httpHeaders:
+          actions:
+            response:
+            - name: X-Frame-Options
+              action:
+                type: Set
+                set:
+                  value: DENY
+            - name: X-Cache-Info
+              action:
+                type: Set
+                set:
+                  value: "not cacheable; meta data too large"      
+            - name: X-XSS-Protection
+              action:
+                type: Delete
+            request:
+            - name: Content-Location
+              action:
+                type: Set
+                set:
+                  value: /my-first-blog-post
+            - name: X-SSL-Client-Cert 
+              action:
+                type: Set
+                set:
+                  value: "%{+Q}[ssl_c_der,base64]"
+            - name:   Content-Language
+              action:
+                type: Delete
+            - name: X-Target
+              action:
+                type: Set
+                set:
+                  value: "%[req.hdr(host),lower]"
+    expected: |
+      apiVersion: route.openshift.io/v1
+      kind: Route
+      metadata:
+        labels:
+          type: sharded
+        name: hello-openshift-actions
+        namespace: hello-openshift
+      spec:
+        subdomain: hello-openshift
+        tls:
+          termination: edge
+        to:
+          kind: Service
+          name: hello-openshift
+          weight: 100
+        wildcardPolicy: None      
+        httpHeaders:
+          actions:
+            response:
+            - name: X-Frame-Options
+              action:
+                type: Set
+                set:
+                  value: DENY
+            - name: X-Cache-Info
+              action:
+                type: Set
+                set:
+                  value: "not cacheable; meta data too large"       
+            - name: X-XSS-Protection
+              action:
+                type: Delete
+            request:
+            - name: Content-Location
+              action:
+                type: Set
+                set:
+                  value: /my-first-blog-post
+            - name: X-SSL-Client-Cert 
+              action:
+                type: Set
+                set:
+                  value: "%{+Q}[ssl_c_der,base64]"
+            - name:   Content-Language
+              action:
+                type: Delete
+            - name: X-Target
+              action:
+                type: Set
+                set:
+                  value: "%[req.hdr(host),lower]"
+  - name: Should not allow actions if tls termination is set to passthrough
```

Comment:
This test needs to be split into two to make sure this is working correctly, please add a "Should not allow request header actions if tls termination is set to passthrough" and "Should not allow response header actions if tls termination is set to passthrough" so that we test each of the two sets individually

Reply:
i think this split really does not matter as we are not checking this per response or request.

Reply:
anyway added the split

Reply:
We are, the previous version of the validation wouldn't have failed on these two tests split apart, that's why I've asked for it